datafusion/functions/src/unicode/substr.rs

```diff
@@ -89,6 +100,63 @@ impl ScalarUDFImpl for SubstrFunc {
     }
 }
 
+// TODO(aduffy): just return a manipulated Utf8View array by modifying the view only.
+pub fn substr_view(args: &[ArrayRef]) -> Result<ArrayRef> {
+    match args.len() {
```

Comment:
I'm going to revert this since it's outside the scope of this PR.

Ideally, we'd be able to defer to the `substring` kernel in arrow_string, but that kernel needs to be fixed to support utf8views